spec/models/ticket_scanning_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'spec_helper'
+
+describe TicketScanning do
+  let(:conference) { create(:conference) }
+  let(:user) { create(:user) }
+  let(:registration) { create(:registration, conference: conference, user: user) }
+  let(:registration_ticket) { create(:registration_ticket, conference: conference) }
+  let(:paid_ticket_purchase) { create(:ticket_purchase, conference: conference, user: user, ticket: registration_ticket) }
+  let(:physical_ticket) { create(:physical_ticket, ticket_purchase: paid_ticket_purchase) }
+  let(:ticket_scanning) { create(:ticket_scanning, physical_ticket: physical_ticket) }
+
+  describe 'before_create' do
+    it 'marks user as present' do
+      expect(registration.attended).to eq(false)
+      ticket_scanning
+      registration.reload
+      expect(registration.attended).to eq(true)
```

Comment:
you should have two expect in the same test, split the test :wink: 

http://www.betterspecs.org/#single

Reply:
Nice resource. 😉 
But here both expect are for the same behaviour so they must be present in the same test.

Reply:
They are not, one is to test that by default the user attendance is false, and the other to check that is set to true afterwards.  And actually the false part is a requirement, more than something that should be tested. Anyway, let's merge this and we can improve the test afterwards. It is at least tested.